File: src/main/java/io/github/nikpivkin/events/LifetimeChangedEvent.java
package io.github.nikpivkin.events;

import io.github.nikpivkin.Lifetime;
import org.bukkit.entity.Player;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

public class LifetimeChangedEvent extends Event {

  private static final HandlerList HANDLERS = new HandlerList();
  private final Player player;
  private final Lifetime oldValue;
  private final Lifetime newValue;

  public LifetimeChangedEvent(Player player, Lifetime oldValue, Lifetime newValue) {
    this.player = player;
    this.oldValue = oldValue;
    this.newValue = newValue;
  }

  @Override
  public @NotNull HandlerList getHandlers() {
    return HANDLERS;
  }

  public static HandlerList getHandlerList() {
    return HANDLERS;
  }

  public Player getPlayer() {
    return player;
  }

  public Lifetime getOldValue() {
    return oldValue;
  }

  public Lifetime getNewValue() {
    return newValue;
  }
}

File: src/main/kotlin/io/github/nikpivkin/lifetime/events/LifetimeChangedEvent.kt
package io.github.nikpivkin.lifetime.events;

import io.github.nikpivkin.lifetime.Lifetime
import org.bukkit.entity.Player
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

data class LifetimeChangedEvent(
  val player: Player,
  val oldValue: Lifetime,
  val newValue: Lifetime
): Event() {

  override fun getHandlers() = HANDLERS

  companion object {
    private val HANDLERS = HandlerList();

    @JvmStatic
    fun getHandlerList() = HANDLERS
  }
}
